walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,140 @@
+   *         available.
+   */
+  private ArrayList<ArrayList<TimeRange>> getAllAttendeeAvailabilities(
+      Collection<Event> events, Collection<String> attendees) {
+    ArrayList<ArrayList<TimeRange>> attendeeAvailabilities = new ArrayList<ArrayList<TimeRange>>();
+    for (String attendee : attendees) {
+      attendeeAvailabilities.add(getAttendeeAvailability(events, attendee));
+    }
+    return attendeeAvailabilities;
+  }
+
+  /**
+   * Gets the available times in a day that the given attendee is available.
+   *
+   * @param events {@code Event} Collection for the day.
+   * @param attendee Person who we wish to find the available times.
+   * @return {@code TimeRange} ArrayList where the attendee is available.
+   */
+  private ArrayList<TimeRange> getAttendeeAvailability(Collection<Event> events, String attendee) {
+    ArrayList<TimeRange> availableTimes = new ArrayList<TimeRange>();
+
+    // Add TimeRanges over the entire day, skipping over the events that contains the attendee.
+    int availableStart = TimeRange.START_OF_DAY;
+    for (Event event : events) {
+      if (event.getAttendees().contains(attendee)) {
+        int eventStart = event.getWhen().start();
+        int eventEnd = event.getWhen().end();
+
+        availableTimes.add(TimeRange.fromStartEnd(availableStart, eventStart, false));
+        availableStart = eventEnd;
+      }
+    }
+    availableTimes.add(TimeRange.fromStartEnd(availableStart, TimeRange.END_OF_DAY, true));
+
+    return availableTimes;
+  }
+
+  /**
+   * Given a {@code TimeRange} list of lists, return the interval list intersections.
+   *
+   * @param allLists The {@code TimeRange} list of lists that we wish to get the intersection of.
+   * @return The intersection of all the lists given.
+   */
+  private ArrayList<TimeRange> allTimeRangesIntersection(ArrayList<ArrayList<TimeRange>> allLists) {
+    ArrayList<TimeRange> availableTimes = new ArrayList<TimeRange>();
+    // Whole day TimeRange is a base case because it is the identity element, and should also be
+    // returned when no attendees are given.
+    availableTimes.add(TimeRange.WHOLE_DAY);
+    for (ArrayList<TimeRange> ranges : allLists) {
+      availableTimes = twoTimeRangesIntersection(availableTimes, ranges);
+    }
+    return availableTimes;
+  }
+
+  /**
+   * Given two {@code TimeRange} ArrayLists, returns the intersection of the two interval lists.
+   *
+   * Example: |---| |--|
+   *            |-----|
+   * Returns:   |-| |-|
+   *
+   * @param arr1 The first ArrayList to intersect with.
+   * @param arr2 The second ArrayList to intersect with.
+   * @return The intersection of the two ArrayLists.
+   */
+  private ArrayList<TimeRange> twoTimeRangesIntersection(
+      ArrayList<TimeRange> arr1, ArrayList<TimeRange> arr2) {
+    ArrayList<TimeRange> intersection = new ArrayList<TimeRange>();
+    int arr1Index = 0, arr2Index = 0;
```

Comment:
```suggestion
    int arr1Index = 0;
    int arr2Index = 0;
```
[Style guide](https://google.github.io/styleguide/javaguide.html#s4.8.2-variable-declarations)